File: front-ui/src/api/house.ts
import request from './request';

export interface HouseQuery {
  district?: string;
  minPrice?: number;
  maxPrice?: number;
  minArea?: number;
  maxArea?: number;
  roomCount?: number;
  type?: string;
  status?: string;
}

export interface HouseData {
  id?: number;
  title: string;
  address: string;
  district: string;
  price: number;
  area: number;
  roomCount: number;
  floor: number;
  totalFloor: number;
  buildYear?: number;
  type: string;
  status?: string;
  features?: string[];
  facilities?: string[];
  description?: string;
}

export function getHouseList() {
  return request({
    url: '/houses',
    method: 'get'
  });
}

export function searchHouses(data: HouseQuery) {
  return request({
    url: '/houses/search',
    method: 'post',
    data
  });
}

export function getHouseById(id: number) {
  return request({
    url: `/houses/${id}`,
    method: 'get'
  });
}

export function createHouse(data: HouseData) {
  return request({
    url: '/houses',
    method: 'post',
    data
  });
}

export function updateHouse(id: number, data: HouseData) {
  return request({
    url: `/houses/${id}`,
    method: 'put',
    data
  });
}

export function deleteHouse(id: number) {
  return request({
    url: `/houses/${id}`,
    method: 'delete'
  });
} 

File: src/main/java/com/demo/javademo/entity/House.java
package com.demo.javademo.entity;

import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class House {
    private Long id;
    private String title;
    private String address;
    private String district;
    private BigDecimal price;
    private BigDecimal area;
    private Integer roomCount;
    private Integer floor;
    private Integer totalFloor;
    private String buildYear;
    private String type;
    private String status;
    private String features;
    private String facilities;
    private String description;
    private Long creatorId;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
} 